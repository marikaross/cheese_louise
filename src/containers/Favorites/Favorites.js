import React from 'react';
import { CheeseCard } from '../../components/CheeseCard';
import { connect } from 'react-redux';
import { addFaveCheese, deleteCheese } from '../../actions';


export const Favorites = (props) => {

   const toggleFave = (id) => {
    if(isDuplicate(id)) {
      props.deleteCheese(id)
    } else {
      props.addFaveCheese(id)
    }
  }

   const isDuplicate = (id) => {
    return props.favorites.find(favoriteId => favoriteId === id)
  }

  const faveCheeses = props.favorites.reduce((faveCheeses,favorite) => {
    const allCheese = props.cheeses.filter(cheese => {
      return cheese.cheeseId === favorite
    })
    return [...faveCheeses, ...allCheese]
  },[])



  const cheeseCards = faveCheeses.map(cheese => {
    return (
      <CheeseCard
        name={cheese.name}
        milk={cheese.milk}
        region={cheese.region}
        id={cheese.cheeseId}
        toggleFave={toggleFave}
        />
      )
  })

  return (
    (cheeseCards)
    )

}

export const mapStateToProps = (state) => ({
  cheeses: state.cheeses,
  favorites: state.favorites
})

export const mapDispatchToProps = (dispatch) => ({
  deleteCheese: (id) => dispatch(deleteCheese(id)),
  addFaveCheese: (id) => dispatch(addFaveCheese(id))
})

export default connect(mapStateToProps, mapDispatchToProps)(Favorites);