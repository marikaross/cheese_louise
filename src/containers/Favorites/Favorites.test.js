import { Favorites } from './Favorites';
import React from 'react';
import { shallow } from 'enzyme';

describe('Favorites', () => {
  let wrapper
  let mockCheeses
  let mockFavorites
  let mockDeleteCheese
  let mockAddFaveCheese

  beforeEach(() => {
    mockCheeses = [{name: 'nacho', id: 6}, {name: 'swiss', id: 5}]
    mockFavorites = [{name: 'nacho', id: 6}]
    mockDeleteCheese = jest.fn()
    mockAddFaveCheese = jest.fn()

     wrapper = shallow(<Favorites 
      cheeses={mockCheeses}
      favorites={mockFavorites}
      deleteCheese={mockDeleteCheese}
      addFaveCheese={mockAddFaveCheese}/>)
  })

  it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})