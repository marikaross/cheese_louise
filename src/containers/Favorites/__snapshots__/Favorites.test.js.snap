// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match the snapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    addFaveCheese={[Function]}
    cheeses={
        Array [
            Object {
              "id": 6,
              "name": "nacho",
            },
            Object {
              "id": 5,
              "name": "swiss",
            },
          ]
    }
    deleteCheese={[Function]}
    favorites={
        Array [
            Object {
              "id": 6,
              "name": "nacho",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [],
  Symbol(enzyme.__nodes__): Array [
    Array [],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
